tasks/actorstate/miner.go

```diff
@@ -31,16 +33,71 @@ func init() {
 }
 
 func (m StorageMinerExtractor) Extract(ctx context.Context, a ActorInfo, node ActorStateAPI) (model.Persistable, error) {
-	// TODO:
-	// - all processing below can and probably should be done in parallel.
-	// - processing is incomplete, see below TODO about sector inspection.
-	// - need caching infront of the lotus api to avoid refetching power for same tipset.
+	// TODO all processing below can, and probably should, be done in parallel.
 	ctx, span := global.Tracer("").Start(ctx, "StorageMinerExtractor")
 	defer span.End()
 
 	stop := metrics.Timer(ctx, metrics.ProcessingDuration)
 	defer stop()
 
+	ec, err := NewMinerStateExtractionContext(ctx, a, node)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO add this to the power actor processing.
```

Comment:
is this in a subsequent PR?

Reply:
Yes, issue for it here: https://github.com/filecoin-project/sentinel-visor/issues/171. Picking it up next.